src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -365,6 +365,10 @@ request_creator::make_delete_objects_request(
 
         auto out = std::ostringstream{};
         boost::property_tree::write_xml(out, delete_tree);
+        if (!out.good()) {
+            throw std::runtime_error(
+              fmt_with_ctx(fmt::format, "failed to create delete request"));
```

Comment:
should we print out the iostate flags and other context so we have a better idea of why the failure occurred?

Reply:
added that